### PR TITLE
chore: remove framework patch for custom fields

### PR DESCRIPTION
**Pull request by shadrak98**
While migrating from older versions to the latest versions few patches were breaking.
This patch is moved to framework

### CODE REVIEW

File: erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ def execute():
 			("stock", "purchase_receipt_item"),
 			("stock", "delivery_note_item"),
 			("stock", "packed_item"),
+			("stock", "stock_ledger_entry"),
```
Comment (ankush) [inline erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py]: This is already done in a separate patch. It's not wise to reload it here.

Schema for SLE actually changed between v12 and v13 and hence the patch that migrates schema should only touch it.

**Review by ankush — CHANGES_REQUESTED** [review]

File: erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py (hunk 1 above)
Comment (shadrak98) [reply erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py]: The patches were still breaking during migration.. do we need to reorder the patch then?

Comment (ankush) [reply erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py]: the patch already runs before this one; it's called "update_is_cancelled_field"

Comment (ankush) [reply erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py]: https://github.com/frappe/erpnext/blob/964b911b32d71e3069ad0f942b447cd25cb4da1c/erpnext/patches/v12_0/update_is_cancelled_field.py#L12

Comment (shadrak98) [reply erpnext/patches/v13_0/item_reposting_for_incorrect_sl_and_gl.py]: Ahh, this is strange. Nvm, I will remove it. Thanks!

File: erpnext/patches/v13_0/validate_options_for_data_field.py [hunk 2]
```diff
@@ -9,18 +9,18 @@
...
-                    options=NULL
-                WHERE
-                    name=%s
-            """, (field))
+        if field.options not in data_field_options:
+            custom_field = frappe.qb.DocType('Custom Field')
+            frappe.qb.update(
+                custom_field
+            ).set(
+                custom_field.options, "NULL"
```
Comment (ankush) [inline erpnext/patches/v13_0/validate_options_for_data_field.py]: this is the string `"null"` and not SQL `NULL`. 😄 

It should be `None` IIRC

Comment (ankush) [reply erpnext/patches/v13_0/validate_options_for_data_field.py]: also this whole patch doesn't belong in ERPNext. Framework added this validation, so patch/fix should be in framework.

Comment (shadrak98) [reply erpnext/patches/v13_0/validate_options_for_data_field.py]: Yeah thanks for pointing this out!
Will close this one and raise a PR in FF

Comment (ankush) [reply erpnext/patches/v13_0/validate_options_for_data_field.py]: instead of closing this, reuse it to remove that framework patch? 😅
